### ISSUE TITLE
KmlOverlay OVERLAY_CLICK event not working

### ISSUE THREAD

**Issue by langvtran** [mapsplugin/cordova-plugin-googlemaps#915]
Adding an on-click event listener to KML overly is not activating the call back function. In the sample code, I would expect to see "overlay_click: " + overlay.type in console output.

[kmloverlay-defect.zip](https://github.com/mapsplugin/cordova-plugin-googlemaps/files/193928/kmloverlay-defect.zip)

**Comment by wf9a5m75** [comment]
Thank you, hold on please.

**Comment by jaredgreener** [comment]
The infowindow event 'infoClick' is also not functioning

**Comment by jaredgreener** [comment]
NVM -- looks like the docs changed slightly, you MUST also include:

'markerClick': function(marker) {
    marker.showInfoWindow();
  },

**Comment by wf9a5m75** [comment]
@jaredgreener Where does the code into?
@langvtran askes about KMLLayer.

During I left this project, someone might change the code. I'm not sure.

**Comment by jaredgreener** [comment]
Before I did this and it worked:

nativeMap.addMarker({
				position: currentMarker.position,
				icon: currentMarker.icon,
				doorID: currentMarker.doorID,
				disableAutoPan: true,
				title: title,
				snippet: snippet,
				'markerClick': function(marker) {
					marker.showInfoWindow();
				},
				infoClick: function(marker){
					stayInMap = true;
				    editDoor(marker.get('doorID'));
				}
			}, 
			function(marker) {
				
// 				console.log("finished placing actual marker "+ currentMarker.doorID);
						
				setTimeout(processDoorMarkerQueue, markerDelay);
				doorMarkers.push(marker);
				
				currentMarker.callback && currentMarker.callback();
		    }
		);

Now I have to add the 'markerClick' event as well to make it work:

nativeMap.addMarker({
				position: currentMarker.position,
				icon: currentMarker.icon,
				doorID: currentMarker.doorID,
				disableAutoPan: true,
				title: title,
				snippet: snippet,
				infoClick: function(marker){
					stayInMap = true;
				    editDoor(marker.get('doorID'));
				}
			}, 
			function(marker) {
				
// 				console.log("finished placing actual marker "+ currentMarker.doorID);
						
				setTimeout(processDoorMarkerQueue, markerDelay);
				doorMarkers.push(marker);
				
				currentMarker.callback && currentMarker.callback();
		    }
		);

**Comment by wf9a5m75** [comment]
@jaredgreener That's another talk. Not related with KMLLayer.

**Comment by wf9a5m75** [comment]
@langvtran I can not read your KML file with Android build.

**Comment by wf9a5m75** [comment]
On iOS neither....

**Comment by wf9a5m75** [comment]
Ok, reproduced your issue. The `OVERLAY_CLICK` should work with your code.

**Comment by wf9a5m75** [comment]
@jaredgreener Have you tried on Android? I guess your issue is on iOS.

**Comment by wf9a5m75** [comment]
@langvtran

**Comment by wf9a5m75** [comment]
annoying github... :(
The issue is automatically closed, but not yet.

**Comment by wf9a5m75** [comment]
@langvtran Yes, your issue was fixed.
```
[CDVTimer][TotalPluginStartup] 35.721004ms
THREAD WARNING: ['GoogleMaps'] took '109.829834' ms. Plugin should use a background thread.
HelloCordova[16078:153880] ==== onMapReady
HelloCordova[16078:153880] Google Maps SDK for iOS version: 1.11.21919.0
HelloCordova[16078:153880] New version of Google Maps SDK for iOS available: 1.12.23211.0
HelloCordova[16078:153880] overlay_click: Marker  <---
```

However (for my case) , tricky thing is that you need to install `cordova-plugin-console` in order to output the strings.
`$> cordova plugin add cordova-plugin-console`

**Comment by langvtran** [comment]
Thanks, works now.
